File: declined-node/src/resources/events.ts
import type { HttpClient } from "../client.js";
import { API_PATHS } from "../paths.js";
import type { EventCreateParams } from "../types.js";

export class EventsResource {
  constructor(private readonly http: HttpClient) {}

  create(params: EventCreateParams) {
    return this.http.request("POST", API_PATHS.events, params);
  }

  markPaymentRecovered(params: Omit<EventCreateParams, "type">) {
    return this.create({ ...params, type: "payment_recovered" });
  }
}

File: declined-python/declined/resources/events.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any

from declined.paths import API_PATHS

if TYPE_CHECKING:
    from declined.client import HttpTransport


class EventsResource:
    def __init__(self, http: HttpTransport):
        self._http = http

    def create(self, **params: Any) -> Any:
        return self._http.request("POST", API_PATHS["events"], body=params)

    def mark_payment_recovered(self, **params: Any) -> Any:
        return self.create(type="payment_recovered", **params)
